docs(release-notes): finalize 8.11 release (#5672)
canonical/snapcraft: docs/release-notes/snapcraft-8-11.rst

first positional argument, Snapcraft skips the build process and instead packs the
directory as a snap.

Unified verbosity arguments
~~~~~~~~~~~~~~~~~~~~~~~~~~~

There are five levels of verbosity to choose from when building core22 and core24 snaps:
quiet, brief, verbose, debug, and trace. These can be selected with the command line
arguments ``--quiet``, ``--verbose``,  and ``--verbosity <level>`` or the environment
variable ``CRAFT_VERBOSITY_LEVEL``.

> However, these options weren't supported by all commands or when building core20 snaps.
> This led to an inconsistent user experience.
> 
> Now, these verbosity options can be used for all commands and when building snaps with
> core20, core22, and core24 bases.

Reviewer: ```suggestion
However, these options weren't supported by all commands or when building core20 snaps.
This led to an inconsistent user experience. To remedy this, we've made these verbosity
options available for all commands when building snaps with core20, core22, and
core24 bases.
```
Another subjective one. Up to you!

Minor features
--------------

Contribution guidelines
~~~~~~~~~~~~~~~~~~~~~~~

In an effort to improve the contributor experience, we've updated our `contribution
guidelines <https://github.com/canonical/snapcraft/blob/main/CONTRIBUTING.md>`_ and

> added a new :ref:`contribute-to-this-documentation` page, which explains the philosophy
> behind Snapcraft's documentation and how to get involved.

Reviewer: ```suggestion
added a new :ref:`contribute-to-this-documentation` page, which explains the Snapcraft
documentation's design principles and how to get involved.
```
Small change after coming back to this with fresh eyes.

Author: https://github.com/canonical/snapcraft/pull/5672/commits/01afe1c58038c502997b155441c81f3fc9f387b5

Now, Snapcraft builds two separate snaps, ``hello-world_1.0_rpi.snap`` and
``hello-world_1.0_rpi-debug.snap``.

Removed core20 error reporting
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When an internal error occurred while building a core20 snap, Snapcraft would prompt
users to submit an anonymous report to Sentry.

This has been removed in favor of submitting bug reports to Snapcraft's `GitHub issues
<https://github.com/canonical/snapcraft/issues>`_ page.

> Accordingly, the environment variables ``SNAPCRAFT_ENABLE_ERROR_REPORTING`` and
> ``SNAPCRAFT_ENABLE_SILENT_REPORT`` are no longer used.

Reviewer: ```suggestion
As a result, the ``SNAPCRAFT_ENABLE_ERROR_REPORTING`` and
``SNAPCRAFT_ENABLE_SILENT_REPORT`` environment variables are no longer in use.
```
Super subjective.

Author: https://github.com/canonical/snapcraft/pull/5672/commits/01afe1c58038c502997b155441c81f3fc9f387b5
